python/ray/tests/test_runtime_env.py

```diff
@@ -167,5 +168,29 @@ def run(runtime_env):
         run(runtime_env)
 
 
+def test_large_runtime_env_fails_fast(start_cluster_shared):
+    """
+    Tests that a task with a runtime_env that is too large fails quickly
+    instead of hanging. This is a regression test for GitHub issue #47432.
+    """
+    cluster, address = start_cluster_shared
+    ray.init(address)
+
+    # Create a runtime_env with a very large environment variable to trigger
+    # a E2BIG error.
+    large_env_vars = {"MY_HUGE_VAR": "X" * 4096 * 100}
+    runtime_env = {"env_vars": large_env_vars}
+
+    @ray.remote
+    def f():
+        # This code should not be reached.
+        return 1
+
+    # The E2BIG error from the raylet is propagated to the
+    # driver, which should raise a RuntimeEnvSetupError.
+    with pytest.raises(RuntimeEnvSetupError):
```

Comment:
```suggestion
    with pytest.raises(RuntimeEnvSetupError, match="<substring of the expected error">):
```
you can verify that the correct error message is propagated this way